src/lightning_app/core/work.py

```diff
@@ -316,9 +316,16 @@ def num_successes(self) -> int:
 
         return has_succeeded_counter
 
+    def _get_property_if_exists(self, name: str) -> Union[property, None]:
+        return getattr(self.__class__, name, None)
+
```

Comment:
since it's just a method calling another function.

Reply:
Just a general question: wouldn't it be slightly more explicit (and convenient to read) if we put the `getattr` under the method? I would really appreciate your thoughts on this! 

@rohitgr7

Reply:
I guess the problem here is that getattr(self.__class__, name, None) can return about anything, doesn't have to be a property. The return type Union[property, None]: wouldn't be appropriate. And you have a check outside with `isinstance(property_object, property)`. This misdirection can be confusing to the reader.

Reply:
Got it, this makes sense! I guess moving the `isinstance(property_object, property)` would be a better solution. Basically, the function would look like this, 
```python
def _get_property_if_exists(self, name: str) -> Union[property, None]:
   attr = getattr(self.__class__, name, None)
   return attr if isinstance(attr, property) else None
```

Reply:
Hi @awaelchli does this sound better? If this solution is better, I could push a quick commit making the change

Reply:
Sure. @rohitgr7 suggestion would also be fine: https://github.com/Lightning-AI/lightning/pull/14259#discussion_r967795208